xcat-inventory/xcclient/inventory/manager.py

```diff
@@ -184,8 +184,17 @@ def validate_args(args, action):
             raise CommandException("Error: the specified directory %(d)s does not exist!", d=args.directory)
         if not os.path.isdir(args.directory):
             raise CommandException("Error: the specified directory %(d)s is not a directory!", d=args.directory)
-    if args.path and os.path.exists(args.path) and not os.path.isfile(args.path):
-            raise CommandException("Error: the specified file %(f)s already exists, is not a file!", f=args.path)
+    if args.path:
+        if os.path.exists(args.path):
+            if not os.path.isfile(args.path):
+                raise CommandException("Error: the specified file %(f)s already exists, is not a file!", f=args.path)
+        else:
+            mydir=os.path.dirname(args.path)
+            myfile=os.path.basename(args.path)
+            if not myfile:
+                raise CommandException("Error: %(f)s is not a file!", f=args.path)
+            if not os.path.exists(mydir):
```

Comment:
I think here, you need using `os.path.isdir`, and message could be as `does not exist or is not a directory`

Reply:
updated